pact/Pact/Core/IR/Eval/CoreBuiltin.hs

```diff
@@ -1229,17 +1279,25 @@ coreIntToStr info b cont handler _env = \case
 
 coreStrToInt :: (CEKEval step b i m, MonadEval b i m) => NativeFunction step b i m
 coreStrToInt info b cont handler _env = \case
-  [VString s] ->
-    checkLen info s *> doBase info cont handler 10 s
+  [VString s] -> do
+    chargeGasArgs info $ GStrOp $ StrOpLength $ T.length s
+    checkLen info s
+    doBase info cont handler 10 s
   args -> argsError info b args
 
 coreStrToIntBase :: (CEKEval step b i m, MonadEval b i m) => NativeFunction step b i m
 coreStrToIntBase info b cont handler _env = \case
   [VInteger base, VString s]
-    | base == 64 -> checkLen info s *> case decodeBase64UrlUnpadded $ T.encodeUtf8 s of
-        Left{} -> throwExecutionError info (DecodeError "invalid b64 encoding")
-        Right bs -> returnCEKValue cont handler $ VInteger (bsToInteger bs)
-    | base >= 2 && base <= 16 -> checkLen info s *> doBase info cont handler base s
+    | base == 64 -> do
+        chargeGasArgs info $ GStrOp $ StrOpLength $ T.length s
```

Comment:
Ditto, I believe we should be charging for parsing.

Reply:
ditto, updated